### DISCUSSION TITLE
Pricing model for self hosted version

### DISCUSSION THREAD

**Discussion by micudaj** [toeverything/AFFiNE#6299] [Q&A] [answered]
Hi,

I noticed that in my self hosted instance there is a limit for users in a workspace. That lead me to the question what the pricing model for self hosted instance is going to be?
Can you please elaborate on that?

Many thanks in advance!

**Comment by EYHN** [comment]
Currently, the free plan is applied by default in selfhost. This is a bug and we will fix it later.

**Comment by micudaj** [comment]
hey @EYHN, Thank you for your reply! What does ist mean that it is a bug? Will it be possible to have more users in a workspace on a self hosted environment without a license? Or will there be special licenses for self hosted environments if one wants to have more that 3 users in a workspace?

**Comment by EYHN** [comment, marked as answer]
@micudaj Sorry for the late reply,

The current selfhost version uses the free plan, which has two limitations.
One is that the maximum number of collaborators in the workspace is 3, and the file storage capacity is limited.

According to our expectations, the selfhost version should have unlimited file storage, and a reasonable maximum number of collaborators. We haven't decided on the number limit yet, but it may be 3 or higher. (enough for personal use)

In addition, we are also preparing a paid selfhost plan for enterprises or studios, which can support more collaborators and we can make some profits.

**Reply by micudaj** [reply]
thank you @EYHN for clarification.

**Reply by gymnae** [reply]
That seems like a reasonable path. Do you have an idea if this would be a recurring fee or a one time, perpetual license? Would it also require the self-hosted installation be able to contact servers for license check?

Is there an ETA for this change? I'd like to onboard some friends to a space for a common hobby, and I'd like to know if it's going to be affordable for a hobbyist / non-profit with more than a dozen users.

**Reply by IanJSaul** [reply]
This is brilliant. I've been following this for a little while, but am now switching over - and these are exactly my questions too.

Sounds like great software and a great team!

**Reply by nauti23** [reply]
@EYHN so i love the app and just wanted to know:
selfhosted < 10 members per workspace is free. if a have 12 i have to pay for the two extra or for all the 12 seats?

**Comment by levx-me** [comment]
I'm facing this issue too 
what's the point self-hosting if I need to pay? 😢

**Comment by agluck91** [comment]
The whole point of self hosting is to get away from the crazy subscription models. This is unfortunate too because this is so far the best note app that I found. Though I understand the need to make a profit. If the product is good enough, maybe consider opening it to pay what you want model, since it cost you nothing in hosting whatever the community can give you in that time is more than 0.

**Comment by wouekk** [comment]
I think the self hosted limitations are reasonable for user count, I mean there is a distinguishable border between self hosted and enterprise. If the tool improves the productivity of your business you should contribute. I started just a few days ago and the limitations dont bother me at all. If you need 20 users you are probably running a company already or you have a very large family. 

You are doing a great job guys I hope you'll keep going!

**Comment by akaSorin** [comment]
Hi folks :)

I really love affine, and I get excited at every feature that comes to us, thanks for the great work!

I'd like to bring a reflection related to the user limits in self-hosted versions (which I fully understand given the project must be financially sustainable)

Some use cases are not enterprise, and yet might need more than the 10 users.

My current use case is a self governing meetup group, without any revenue, that has roughly 20 people. 

Bitwarden also opted to a similar approach affine did in putting some of the features behind a subscription, that sort of makes it feel that even though there is zero recurring costs (assuming we don't update an instance) it requires recurring payments. So Vaultwarden came to be that one alternative to that.

Would id be a reasonable compromise to consider a lifetime workspace payment, for example? Settle Up (an expense tracking app) offers both options, and I think AFFiNE could benefit for such model (which could be exclusive for self-hosting)

You "purchase" a group license at X dollars/euros/yuans and that unlocks a larger member count for one workspace. This way the users contribute to the project development, while not being stuck into a subscription while using its own infrastructure. 
I'd definitely prefer that recurring payments were reserved to where recurring costs occur (like the cloud subscriptions), but for NGOs, non-profits, collectives and so on, these costs are mostly not viable.

Don't get me wrong, I want the project to flourish, and that might be a way to balance adoption and revenue to an extent that benefits grassroots initiatives ;)

**Reply by dylbertram** [reply]
I have continually come back to this same debate probably three times over the last two years. I self-host for platform sustainability. Every time I rule out AFFiNE, even though it has what I see as the best offering, because I feel like there is no path to perpetual. I want to build communities around it. I would be happy to pay for an unlock if the community gets big enough. It's totally reasonable to support the project.

But the prospect of having to switch off the platform because I hit the cap and have to pay 10USD for each user (who may not always even be active in the community) per month for years and years always kills the case. It has all the same issues as a full SaaS company (i.e. Notion and Slack) with the extra steps of paying for and maintaining our own hosting infra as well. 

Just let me give you money one time for the unlock! I am always so disappointed that I can't justify adopting AFFiNE!
